### PR TITLE
Fix for 1165

### LINKED ISSUE #1165: Auth.setPreferredMFA with `NOMFA` doesn't disable MFA
Hello! ✋

I'm currently trying to remove MFA with `Auth.setPreferredMFA(user, mfaMethod)` function.
Setting SMS or TOTP works just fine... But when I set `mfaMethod` to `NOMFA`, I still got to enter code received by SMS.

I'm I doing something wrong here ?
Any ideas ?

### LINKED ISSUE #1204: Add GetUserData to obtain PreferredMfaSetting
Do you want to request a feature or report a bug?
feature
What is the current behavior?
When MFA is enabled with TOTP, the `getMFAOptions` returns `undefined`.
https://github.com/aws/aws-amplify/issues/360

However I need to know whether the MFA is enabled... I am able to get `PreferredMfaSetting: "SOFTWARE_TOKEN_MFA"` from a call to `cognitoUser.getUserData`
https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_GetUser.html

or use case 31 in 
https://github.com/aws/aws-amplify/tree/master/packages/amazon-cognito-identity-js

but there is no respective method in amplify so would be nice to have it, or enhance the `getMFAOptions`.

What is the expected behavior?
getMFAOptions returns info about TOTP.

### LINKED ISSUE #1259: MFA SMS & TOTP: 'currentAuthenticatedUser' response differing
### Do you want to request a feature or report a bug?

**Bug** 🐛 

### What is the current behavior?

PATH 1: `Auth.setPreferredMFA(user, SMS)`
PATH 2: `Auth.setPreferredMFA(user, TOTP)`

when we run `Auth.currentAuthenticatedUser()` after these flows to get the mfa preference.. we get the following response:

PATH 1
```
{  
   "MFAOptions":[  
      {  
         "AttributeName":"phone_number",
         "DeliveryMedium":"SMS"
      }
   ],
   "UserAttributes":[  
      {  
         "Name":"sub",
         "Value":"uuid"
      },
      {  
         "Name":"email_verified",
         "Value":"true"
      },
      {  
         "Name":"name",
         "Value":"chris"
      },
      {  
         "Name":"phone_number_verified",
         "Value":"false"
      },
      {  
         "Name":"phone_number",
         "Value":"XXX"
      },
      {  
         "Name":"custom:signupdomain",
         "Value":"http://localhost:9001"
      },
      {  
         "Name":"email",
         "Value":"chris@example.com"
      }
   ],
   "Username":"UUID"
}
```

PATH 2:

```
{
   "PreferredMfaSetting":"SOFTWARE_TOKEN_MFA",
   "UserAttributes":[
      {
         "Name":"sub",
         "Value":"uuid"
      },
      {
         "Name":"email_verified",
         "Value":"true"
      },
      {
         "Name":"name",
         "Value":"chris"
      },
      {
         "Name":"custom:signupdomain",
         "Value":"http://localhost:9001"
      },
      {
         "Name":"email",
         "Value":"chris@example.com"
      }
   ],
   "UserMFASettingList":[
      "SOFTWARE_TOKEN_MFA"
   ],
   "Username":"uuid"
}
```

Surely both should follow the same response format? they are currently completely different.

What adds even more complexity to this is if I now set SMS after I've enabled TOTP, I get the following response:

```
{  
   "PreferredMfaSetting":"SMS_MFA",
   "UserAttributes":[  
      {  
         "Name":"sub",
         "Value":"uuid"
      },
      {  
         "Name":"email_verified",
         "Value":"true"
      },
      {  
         "Name":"name",
         "Value":"chris"
      },
      {  
         "Name":"phone_number_verified",
         "Value":"false"
      },
      {  
         "Name":"phone_number",
         "Value":"xxx"
      },
      {  
         "Name":"custom:signupdomain",
         "Value":"http://localhost:9001"
      },
      {  
         "Name":"email",
         "Value":"chris@example.com"
      }
   ],
   "UserMFASettingList":[  
      "SMS_MFA"
   ],
   "Username":"uuid"
}
```

So now we have 1 flow for SMS, 1 flow for TOTP... 2 completely different responses.. If I overwrite the TOTP path with SMS.. I get the correct response back for both flows.. but not the initial SMS setup (PATH1)..

`Auth.getMFAOptions` also seems to be broken for TOTP options returning too as mentioned in https://github.com/aws-amplify/amplify-js/issues/1204, so currently there isn't a standardised way of retrieving the MFA selection for a given authed user, right? I wouldn't say it's feature request, more of a bug.

### What is the expected behavior?

`Auth.currentAuthenticatedUser()` (or `getMFAOptions`) should return the same payload for either SMS or TOTP MFA setup

### Which versions of Amplify, and which browser / OS are affected by this issue? Did this work in previous versions?

1.0+

You can turn on the debug mode to provide more info for us by setting ```window.LOG_LEVEL = 'DEBUG';``` in your app.

**Linked Bugs**
https://github.com/aws-amplify/amplify-js/issues/1226
https://github.com/aws-amplify/amplify-js/issues/1204

### PR DESCRIPTION
**Pull request by powerful23**
*Issue #, if available:*
fixes #1165 
fixes #1204 
fixes #1259 
Adds an api to get current preferred mfa type.
*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/src/Auth.ts [hunk 1]
```diff
@@ -403,6 +403,7 @@ export default class AuthClass {
 
     /**
      * get user current preferred mfa option
+     * this method doesn't work with totp, we need to deprecate it.
      * @param {CognitoUser} user - the current user
      * @return - A promise resolves the current preferred mfa option if success
```
Comment (iarroyo5) [inline packages/auth/src/Auth.ts]: `@deprecated`

**Comment by MiroslavPetrik** [comment]
@powerful23 actually when the MFA is disabled, the `getUserData` response does not have key `PreferredMfaSetting`. Therefore it would be cool to add

`const preferredMFA = data.PreferredMfaSetting || 'NOMFA';`

in the `getPreferredMFA` method.

**Comment by powerful23** [comment]
@MiroslavPetrik  Thanks, will update it.

**Comment by MiroslavPetrik** [comment]
@powerful23 when do you plan to publish this?

**Comment by powerful23** [comment]
@MiroslavPetrik I am going to do a publish this week.

**Pull request merged**